dockstore-webservice/src/main/java/io/dockstore/webservice/resources/RefererHeaderLengthLimitFilter.java

```diff
@@ -0,0 +1,39 @@
+package io.dockstore.webservice.resources;
+
+import java.util.List;
+
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+
+import io.dockstore.webservice.CustomWebApplicationException;
+import org.apache.http.HttpStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Every request will be filtered through here to determine if the length of the referer header does exceed our limit.
+ * This is done for two reasons:
+ * 1. Prevent very long query parameters. For Dockstore, these should be relatively short for the most part.
+ * 2. The search page/elastic search will begin fail if the search query becomes too long. Although these appear to be query parameters, the UI
```

Comment:
Interesting, I didn't understand this until I read your comment. I had assumed these were URL query parameters the API is receiving.

Based on your comment, I don't think the filter is the right approach. Sorry, I had suggested it, but I suggested it under the assumption that we were talking about the URL of the endpoint.

1.  It will only catch requests sent from the browser. I think the CLI lets you search, and users could invoke the API directly -- it wouldn't catch those.
2. Relying on the referer header seems too fragile to me. Maybe some browser doesn't implement it correctly, maybe there's a legitimate case for another endpoint where the referer is greater than 500 and shouldn't be blocked.

I think you should add a check instead in [ToolsApiExtendedServiceImpl](https://github.com/dockstore/dockstore/blob/ed21c90690b52d56dbe9731325488f4533cf5ade/dockstore-webservice/src/main/java/io/dockstore/webservice/resources/proposedGA4GH/ToolsApiExtendedServiceImpl.java#L210) -- check the number of keys and/or the accumulated length of the values.

Reply:
> It will only catch requests sent from the browser.

Could you elaborate on this?